File: SearchInsertPosition.cpp
#include <iostream>
using namespace std;

class Solution 
{
public:
    int searchInsert(int A[], int n, int target) 
    {
		int mid, start, end;
		start = 0;
		end = n - 1;
		//mid = (start + end) / 2;
		while (start <= end)
		{
			mid = (start + end) / 2;
			if (A[mid] == target) return mid;
			if (A[mid] > target) end = mid - 1;
			else start = mid + 1;
		}
		if (A[mid] > target) return mid;
		else return mid + 1;        
    }
};
int main()
{

	return 0;
}

File: JavaSolution/src/SearchInsertPosition.java
/**
 * Created by xyunpeng on 2/3/16.
 */
public class SearchInsertPosition {
    public int searchInsert(int[] nums, int target) {
        if (nums == null || nums.length == 0) {
            return 0;
        }

        int start = 0;
        int end = nums.length - 1;
        int mid;
        while (start + 1 < end) {
            mid = start + ((end - start) >> 1);
            if (nums[mid] == target) {
                return mid;
            } else if (nums[mid] > target) {
                end = mid;
            } else if (nums[mid] < target) {
                start = mid;
            }
        }

        if (target <= nums[start]) {
            return start;
        } else if (target <= nums[end]) {
            return end;
        } else {
            return end + 1;
        }
    }
}
